src/library.js

```diff
@@ -1527,6 +1527,7 @@ LibraryManager.library = {
   longjmp__sig: 'vii',
 #if SUPPORT_LONGJMP
   longjmp: function(env, value) {
+    {{{ from64('env') }}};
```

Comment:
Looking at this I think we can maybe avoid this change by moving `longjmp` itself into have code and then renaming this functions to just be `_emscripten_throw_longjmp`:

```
_emscripten_throw_longjump: function() { throw 'longjmp'; }
```

Reply:
That sounds really good if possible. But not sure how to resolve this case. 

There are two cases we generate a call to `emscripten_longjmp` in WebAssemblyLowerEmscriptenEHSjLj pass in LLVM;
1. Replace C `longjmp` call with `emscripten_longjmp`: [Detail link](https://github.com/llvm/llvm-project/blob/88a48999d249a5478d813596d1cfac6ba82126dc/llvm/lib/Target/WebAssembly/WebAssemblyLowerEmscriptenEHSjLj.cpp#L126-L129)
2. Transform every call that might longjmp into some complex code sequence, when `emscripten_longjmp` is [called](https://github.com/llvm/llvm-project/blob/88a48999d249a5478d813596d1cfac6ba82126dc/llvm/lib/Target/WebAssembly/WebAssemblyLowerEmscriptenEHSjLj.cpp#L165) somewhere in the middle of it: [Detail link](https://github.com/llvm/llvm-project/blob/88a48999d249a5478d813596d1cfac6ba82126dc/llvm/lib/Target/WebAssembly/WebAssemblyLowerEmscriptenEHSjLj.cpp#L156-L186)

The second case is fine, because we know the two parameters to `emscripten_longjmp` while generating the call. The first case is tricky, because existing `longjmp` calls can be indirect calls, in which case we don't know those two arguments to `longjmp` are. So in that case we just [cast](https://github.com/llvm/llvm-project/blob/88a48999d249a5478d813596d1cfac6ba82126dc/llvm/lib/Target/WebAssembly/WebAssemblyLowerEmscriptenEHSjLj.cpp#L649-L656) the value to the type of `emscripten_longjmp`: (i32, i32). Because we don't know the two parameters, we can't generate `setThrew` and `emscripten_throw_longjmp` sequence there. So with this I think it will be hard to remove `emscripten_longjmp` function. WDYT?

Reply:
I wasn't suggesting changing anything about codegen.   Only that we move this function from js code the c code.  It shouldn't be a change the makes any difference to the calling code.

Reply:
Oh I misunderstood what you said, sorry. Yeah, will do.